src/agent/onefuzz/src/machine_id.rs

```diff
@@ -112,6 +116,42 @@ pub async fn get_os_machine_id() -> Result<Uuid> {
     Ok(Uuid::parse_str(&guid)?)
 }
 
+#[cfg(target_os = "macos")]
+pub async fn get_os_machine_id() -> Result<Uuid> {
+    // getting the machine ID can be done through mach APIs, but for
+    // simplicity, this uses a method originally used in
+    // `cpp_client_telemetry`.
+    // ref: https://github.com/microsoft/cpp_client_telemetry/commit/41315d1b77dcf7a419332b96ea4c7cab43760853
+
+    let output = Command::new("ioreg")
+        .args(&["-d2", "-c", "IOPlatformExpertDevice"])
+        .stdin(Stdio::null())
+        .stderr(Stdio::null())
+        .output()
+        .await
+        .context("ioreg failed to run")?;
+
+    let stdout = String::from_utf8_lossy(&output.stdout);
+
+    for line in stdout.split('\n') {
+        // "IOPlatformUUID" = "GUID_HERE"
+        if line.contains("IOPlatformUUID") {
+            let quoted = line.split(" = ").last().ok_or_else(|| {
+                anyhow!("unable to find machine_id from IOPlatformUUID: {}", line)
+            })?;
```

Comment:
Are sure that a line contains `IOPlatformUUID` iff it is in the format we're trying parse?

Reply:
What are you asking?

Reply:
We invoke `ioreg`. We expect one of the output lines to contain a string of the form `"IOPlatformUUID" = "<some-uuid>"`, with quotes around both key and value. If a line contains the key string (`IOPlatformUUID`, not quoted), then we assume it is the k/v line, and (strictly) parse it as such. If parsing fails on any line we try to parse, we bail.

Since we fail fast, if the output of `ioreg` ever emitted a line with the substring `IOPlatformUUID` _before_ the k/v line we care about, then our parse could fail, even if the line we want occurs later.

I'm assuming that's not what happens in practice where you tested, but for all I know, it is possible, which is why I'm asking.

To some extent, this is why I asked about a unit test: the input would be a snapshot of the expected output of `ioreg`, and then the parsing function (which is pure) can be applied against it. 

I don't think this is necessary, but if we wanted to be defensive, would could have a "what if it changed" test case that simulates the condition I just mentioned. This would force us to change the parsing implementation to stop parsing on first success, but keep trying on the remaining output lines if parsing fails for some line with a matching substring.

Reply:
We verify the functionality works as a whole.  I don't think the additional complexity buys us much.

Reply:
That's fine, your call!